execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -915,4 +966,56 @@ where
         self.validate_uref(&contract_package.access_key())?;
         Ok(contract_package)
     }
+
+    /// Safely charge the specified amount of gas, up to the available gas limit.
+    ///
+    /// Returns [`Error::GasLimit`] if gas limit exceeded and `()` if not.
+    /// Intuition about the return value sense is to answer the question 'are we
+    /// allowed to continue?'
+    pub(crate) fn charge_gas(&mut self, amount: Gas) -> Result<(), Error> {
+        let prev = self.gas_counter();
+        let gas_limit = self.gas_limit();
+        // gas charge overflow protection
+        match prev.checked_add(amount) {
+            None => {
+                self.set_gas_counter(gas_limit);
+                Err(Error::GasLimit)
+            }
+            Some(val) if val > gas_limit => {
+                self.set_gas_counter(gas_limit);
+                Err(Error::GasLimit)
+            }
+            Some(val) => {
+                self.set_gas_counter(val);
+                Ok(())
+            }
+        }
+    }
+
+    /// Charges gas for specified amount of bytes used.
+    fn charge_gas_storage(&mut self, bytes_count: usize) -> Result<(), Error> {
+        let storage_costs = self.protocol_data().wasm_config().storage_costs();
+
+        let gas_cost = storage_costs.calculate_gas_cost(bytes_count);
+
+        self.charge_gas(gas_cost)
+    }
+
+    /// Writes data to global state with a measurement
+    pub(crate) fn metered_write_gs_unsafe<K, V>(&mut self, key: K, value: V) -> Result<(), Error>
```

Comment:
This method is awfully lonely and separated from other like methods.  

Possible that we could move it, along with `charge_gas` and `charge_gas_storage` higher up in the `impl` alongside of like methods?

Reply:
Moved further up in the code: charge gas, charge gas storage, and then unsafe write, metered write , metered add unsafe, metered add